paasta_tools/utils.py

```diff
@@ -2255,16 +2256,27 @@ def get_service_instance_list_no_cache(
     for srv_instance_type in instance_types:
         conf_file = f"{srv_instance_type}-{cluster}"
         log.info(f"Enumerating all instances for config file: {soa_dir}/*/{conf_file}.yaml")
-        instances = service_configuration_lib.read_extra_service_information(
-            service,
-            conf_file,
-            soa_dir=soa_dir,
-        )
-        for instance in instances:
-            if instance.startswith('_'):
-                log.debug(f"Ignoring {service}.{instance} as instance name begins with '_'.")
-            else:
-                instance_list.append((service, instance))
+        if srv_instance_type == 'tron':
+            tron_config_content = service_configuration_lib.read_extra_service_information(
+                service,
+                conf_file,
+                soa_dir=soa_dir,
+            )
+            for job in tron_config_content.get('jobs', []) or []:
+                for action in job['actions']:
```

Comment:
Hm, this doesn't include cleanup actions... Same with load_tron_instance_config.
They're annoying to deal with, maybe worth a refactor of how that works later.